compact_str/src/tests.rs

```diff
@@ -695,3 +705,131 @@ fn test_to_compact_string() {
         format_compact!("0{}67890123456789{}", "12345", 999999)
     );
 }
+
+#[test]
```

Comment:
love the existing test coverage, thank you!

Could you add one more test for the 16MB edge case that exists for 32-bit machines? see [here](https://github.com/ParkMyCar/compact_str/blob/main/compact_str/src/repr/boxed/mod.rs#L608) for an example